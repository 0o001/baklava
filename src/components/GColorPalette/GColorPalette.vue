<template>
  <GBox
    :mb="30"
    align="items-center"
  >
    <GBox
      flex
      v-if="palette.name"
      class="g-color-palette"
      direction="column"
    >
      <GBox
        flex
        :pb="20"
      >
        <GText
          color="main-grey-500"
          variant="headline-03"
        >
          {{ palette.name }}
        </GText>
      </GBox>
      <GBox flex>
        <GBox
          v-for="(color,index) in palette.colors"
          :key="index"
        >
          <GBox
            :class="`g-bg-${palette.tag}-${index+1}00`"
            class="color-box"
            :mr="8"
          />
          <GBox :mt="10">
            <GText>
              {{ palette.tag }}-{{ index + 1 }}00
            </GText>
            <GText class="color-hex" color="mid-grey-500" variant="caption">
              {{ color }}
            </GText>
          </GBox>
        </GBox>
      </GBox>
    </GBox>
  </GBox>
</template>

<script>

import GBox from '../GBox/GBox.vue';
import GIcon from '../GIcon/GIcon.vue';
import GText from '../GText/GText.vue';

export default {
  name: 'GColorPalette',
  components: {GText, GBox, GIcon },
  props: {
    palette: {
      type: Object,
      default: {},
    },
  },
};
</script>

<style lang="scss" scoped>
  .g-color-palette {
    width: 100%;
    border-radius: 4px;
    font-size: var(--font-size-12);

    .color{
      &-box{
        height: 70px;
        width: 100px;
      }
      &-hex{
        margin-top: 5px;
      }
    }
  }
</style>
