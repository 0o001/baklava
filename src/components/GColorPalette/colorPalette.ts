const colorPalette = [
  { name: 'Main Grey',
    tag: 'main-grey',
    colors: ['#020203',
      '#0f131a',
      '#1f2734',
      '#171d27',
      '#273142',
      '#525a68',
      '#7d838e',
      '#a9adb3',
      '#d4d6d9',
    ],
  },
  { name: 'Orange',
    tag: 'orange',
    colors: ['#e71805',
      '#ea300a',
      '#ec490f',
      '#ef6114',
      '#f27a1a',
      '#f59548',
      '#f7af76',
      '#facaa3',
      '#fce4d1',
    ],
  },
  { name: 'Dark Grey',
    tag: 'dark-grey',
    colors: ['#161727',
      '#2c2f3f',
      '#424757',
      '#585f6f',
      '#6e7787',
      '#8b929f',
      '#a8adb7',
      '#c5c9cf',
      '#e2e4e7'],
  },
  { name: 'Mid Grey',
    tag: 'mid-grey',
    colors: ['#405579',
      '#556888',
      '#6a7b97',
      '#7f8ea6',
      '#95a1b5',
      '#aab4c4',
      '#bfc7d3',
      '#d5d9e1',
      '#eaecf0'],
  },
  { name: 'Light Grey',
    tag: 'light-grey',
    colors: ['#6f849f',
      '#7f92aa',
      '#8f9fb4',
      '#9fadbf',
      '#afbbca',
      '#bfc9d5',
      '#cfd6df',
      '#dfe4ea',
      '#eff1f4'],
  },
  { name: 'Background Grey',
    tag: 'bg-grey',
    colors: ['#d5d8e2',
      '#dddfe9',
      '#e4e5ed',
      '#eeeff5',
      '#f1f2f7',
      '#f4f5f9',
      '#f7f7fa',
      '#f9fafc',
      '#fcfcfd'] },
  { name: 'Green',
    tag: 'green',
    colors: ['#028f12',
      '#049b24',
      '#06a837',
      '#08b449',
      '#0bc15c',
      '#3ccd7d',
      '#6dda9d',
      '#9de6be',
      '#cef3de'],
  },
  { name: 'Red',
    tag: 'red',
    colors: ['#f60000',
      '#ff100a',
      '#ff3028',
      '#ff4035',
      '#ff5043',
      '#ff7369',
      '#ff968e',
      '#ffb9b4',
      '#ffdcd9'],
  },
  { name: 'Purple',
    tag: 'purple',
    colors: ['#300fff',
      '#471fff',
      '#5e2eff',
      '#753eff',
      '#8c4eff',
      '#a371ff',
      '#ba95ff',
      '#d1b8ff',
      '#e8dcff'],
  },
  { name: 'Blue',
    tag: 'blue',
    colors: ['#113eff',
      '#2253ff',
      '#3469ff',
      '#457eff',
      '#5794ff',
      '#79a9ff',
      '#9abfff',
      '#bcd4ff',
      '#ddeaff'],
  },
  { name: 'Yellow',
    tag: 'yellow',
    colors: ['#ff7b00',
      '#ff8a00',
      '#ff9800',
      '#ffa700',
      '#ffb600',
      '#ffc533',
      '#ffd366',
      '#ffe299',
      '#fff0cc'],
  },
];

export default colorPalette;
