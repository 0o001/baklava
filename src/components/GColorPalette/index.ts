import GColorPalette from './GColorPalette.vue';

export default GColorPalette;
