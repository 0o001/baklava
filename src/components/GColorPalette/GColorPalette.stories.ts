import { storiesOf } from '@storybook/vue';
import ColorPalette from './GColorPalette.vue';
import GText from '../GText/GText.vue';
import GBox from '../GBox/GBox.vue';
import GIcon from '../GIcon/GIcon.vue';
import colorPalette from "./colorPalette";
import { withKnobs } from '@storybook/addon-knobs';
const stories = storiesOf('Core|Colors', module);

stories.addDecorator(withKnobs);
stories
  .addParameters({ component: ColorPalette })
  .add('Default', () => ({
    data () {
      return {
        allColors : colorPalette,
      }
    },
    components: { ColorPalette, GIcon, GText, GBox },
    template: `
      <GBox
        class="g-col-lg-shift-2 g-col-lg-8"
        :mt="30"
      >
        <GBox :pb="30">
          <GText
            color="main-grey-500"
            variant="headline-02"
          >
            Colors
          </GText>
          <GBox :py="30">
            <span class="g-fs-18 g-text-main-grey-700">
              Choose colors for UI elements such as labels, text,
              backgrounds, and links.
            </span>
          </GBox>
          <GBox flex direction="column">
            <GText
              color="main-grey-500"
              variant="headline-03"
            >
              Usage
            </GText>
            <span class="g-fs-18 g-text-main-grey-700 g-pt-20">
              To apply a color to an element, apply the following classes rules. The actual color for
              each modifier is defined by the UIkit style that you have chosen.
            </span>
            <span class="g-fs-18 g-text-main-grey-700 g-pt-20">
              Background class rule: g-bg-{tag-name}
            </span>
            <GBox
              flex
              :mt="20"
            >
              <span
                class="g-bg-red-500 g-text-bg-grey-900 g-px-10 g-py-28"
              >
                g-bg-red-500
              </span>
            </GBox>
            <span class="g-fs-18 g-text-main-grey-700 g-pt-20">
              Text class rule: g-text-{tag-name}
            </span>
            <span
              class="g-text-red-500 g-pt-20"
            >
              g-text-red-500
            </span>
          </GBox>
        </GBox>
        <ColorPalette
          v-for="(colors,index) in allColors"
          :palette="colors"
          :key="index"
        />
      </GBox>`,
}));
