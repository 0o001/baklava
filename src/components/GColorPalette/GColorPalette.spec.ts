import {mount, shallowMount, ThisTypedShallowMountOptions, Wrapper} from '@vue/test-utils';
import GColorPalette from './';
import colorPalette from "@/components/GColorPalette/colorPalette";

describe('GColorPalette', () => {
  let wrapper: Wrapper<GColorPalette>;
  const renderWrapper = (options: ThisTypedShallowMountOptions<GColorPalette> = {}) => {
    wrapper = mount(GColorPalette, options);
  };

  it('should match snapshot with palette prop', () => {
    renderWrapper({
      propsData: {
        palette: colorPalette[0],
      },
    })
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should not rendered color palette if palette props does not given', () => {
    renderWrapper({
      propsData: {
        palette: {},
      },
    })
    const wrapperDiv = wrapper.find('div.g-color-palette')
    expect(wrapperDiv.exists()).toBe(false)
  });

  it('should render color palette with color', () => {
    renderWrapper({
      propsData: {
        palette: colorPalette[0],
      },
    })
    const wrapperDiv = wrapper.find('div.g-color-palette')
    expect(wrapperDiv.exists()).toBe(true)
  });
});
