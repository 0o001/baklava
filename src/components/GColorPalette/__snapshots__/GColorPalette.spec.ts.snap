// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GColorPalette should match snapshot with palette prop 1`] = `
<div
  class="g-d-align-items-center g-mb-30"
>
  <div
    class="g-color-palette g-d-flex g-d-direction-column"
  >
    <div
      class="g-d-flex g-pb-20"
    >
      <div
        class="g-text-main-grey-500 -headline-03 g-text"
      >
        
        Main Grey
      
      </div>
    </div>
     
    <div
      class="g-d-flex"
    >
      <div
        class=""
      >
        <div
          class="color-box g-mr-8 g-bg-main-grey-100"
        />
         
        <div
          class="g-mt-10"
        >
          <div
            class="g-text-main-grey-500 -body g-text"
          >
            
            main-grey-100
          
          </div>
           
          <div
            class="color-hex g-text-mid-grey-500 -caption g-text"
          >
            
            #020203
          
          </div>
        </div>
      </div>
      <div
        class=""
      >
        <div
          class="color-box g-mr-8 g-bg-main-grey-200"
        />
         
        <div
          class="g-mt-10"
        >
          <div
            class="g-text-main-grey-500 -body g-text"
          >
            
            main-grey-200
          
          </div>
           
          <div
            class="color-hex g-text-mid-grey-500 -caption g-text"
          >
            
            #0f131a
          
          </div>
        </div>
      </div>
      <div
        class=""
      >
        <div
          class="color-box g-mr-8 g-bg-main-grey-300"
        />
         
        <div
          class="g-mt-10"
        >
          <div
            class="g-text-main-grey-500 -body g-text"
          >
            
            main-grey-300
          
          </div>
           
          <div
            class="color-hex g-text-mid-grey-500 -caption g-text"
          >
            
            #1f2734
          
          </div>
        </div>
      </div>
      <div
        class=""
      >
        <div
          class="color-box g-mr-8 g-bg-main-grey-400"
        />
         
        <div
          class="g-mt-10"
        >
          <div
            class="g-text-main-grey-500 -body g-text"
          >
            
            main-grey-400
          
          </div>
           
          <div
            class="color-hex g-text-mid-grey-500 -caption g-text"
          >
            
            #171d27
          
          </div>
        </div>
      </div>
      <div
        class=""
      >
        <div
          class="color-box g-mr-8 g-bg-main-grey-500"
        />
         
        <div
          class="g-mt-10"
        >
          <div
            class="g-text-main-grey-500 -body g-text"
          >
            
            main-grey-500
          
          </div>
           
          <div
            class="color-hex g-text-mid-grey-500 -caption g-text"
          >
            
            #273142
          
          </div>
        </div>
      </div>
      <div
        class=""
      >
        <div
          class="color-box g-mr-8 g-bg-main-grey-600"
        />
         
        <div
          class="g-mt-10"
        >
          <div
            class="g-text-main-grey-500 -body g-text"
          >
            
            main-grey-600
          
          </div>
           
          <div
            class="color-hex g-text-mid-grey-500 -caption g-text"
          >
            
            #525a68
          
          </div>
        </div>
      </div>
      <div
        class=""
      >
        <div
          class="color-box g-mr-8 g-bg-main-grey-700"
        />
         
        <div
          class="g-mt-10"
        >
          <div
            class="g-text-main-grey-500 -body g-text"
          >
            
            main-grey-700
          
          </div>
           
          <div
            class="color-hex g-text-mid-grey-500 -caption g-text"
          >
            
            #7d838e
          
          </div>
        </div>
      </div>
      <div
        class=""
      >
        <div
          class="color-box g-mr-8 g-bg-main-grey-800"
        />
         
        <div
          class="g-mt-10"
        >
          <div
            class="g-text-main-grey-500 -body g-text"
          >
            
            main-grey-800
          
          </div>
           
          <div
            class="color-hex g-text-mid-grey-500 -caption g-text"
          >
            
            #a9adb3
          
          </div>
        </div>
      </div>
      <div
        class=""
      >
        <div
          class="color-box g-mr-8 g-bg-main-grey-900"
        />
         
        <div
          class="g-mt-10"
        >
          <div
            class="g-text-main-grey-500 -body g-text"
          >
            
            main-grey-900
          
          </div>
           
          <div
            class="color-hex g-text-mid-grey-500 -caption g-text"
          >
            
            #d4d6d9
          
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
